lib/src/inline_parser.dart

```diff
@@ -430,6 +452,278 @@ class TagSyntax extends InlineSyntax {
   }
 }
 
+// Link destinations and labels are tricky to parse; the parsing can benefit
+// from keeping some state in an object. This class helps to "parse" a link
+// (after the link text's closing `]` has been parsed). It is called a 'walker'
+// because it contains a reference to the InlineParser which is known
+// throughout this file as the `parser`. Synonymous.
+class _LinkWalker {
+  final InlineParser parser;
+  final TagState state;
+  final LinkSyntax syntax;
+
+  String text;
+  String destination;
+  String title;
+
+  _LinkWalker(this.parser, this.state, this.syntax);
+
+  // Parse an inline link at the current position. Specifically, [parser.pos]
+  // is expected to be pointing at the opening `(`.
+  bool parseInlineLink() {
+    var i = parser.pos + 1;
+    var ch = parser.charAt(i);
+    int destinationIdx;
+    String destination;
+
+    // Loop past the opening whitespace.
+    while (true) {
+      i++;
+      if (i == parser.source.length) {
+        // EOF. Not a link.
+        return false;
+      }
+      ch = parser.charAt(i);
+      switch (ch) {
+        case $space:
+        case $lf:
+        case $cr:
+        case $ff:
+          // Just padding. Move along.
+          continue;
+        default:
+          break;
+      }
+      break;
+    }
+
+    // According to
+    // [CommonMark](http://spec.commonmark.org/0.28/#link-destination):
+    //
+    // > A link destination consists of [...] a nonempty sequence of
+    // > characters [...], and includes parentheses only if (a) they are
+    // > backslash-escaped or (b) they are part of a balanced pair of
+    // > unescaped parentheses.
+    //
+    // Start with 1 for the paren that opened this destination.
+    var parenCount = 1;
+
+    if (ch == $lt) {
+      // Maybe a `<...>`-enclosed link destination.
+      destinationIdx = i + 1;
+      enclosedDestinationLoop:
+      while (true) {
+        i++;
+        ch = parser.charAt(i);
+        switch (ch) {
+          case $backslash:
+            i++;
```

Comment:
The escaping is handled elsewhere, since they want some other weird handling of quotes and such.

But I see your point later on in the escapeAttribute method. Maybe backslash escaping should be done here rather than later.